test/src/test/java/org/axonframework/test/saga/FixtureExecutionResultImplTest.java

```diff
@@ -377,6 +392,148 @@ void testStartRecordingCallback() {
+    }
+
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlineAtTo() {
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(createDeadline(deadlineWindowTo)));
+
+        assertThrows(AxonAssertionError.class, () -> testSubject.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything()));
+    }
+
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlinesOutsideWindow() {
+        ScheduledDeadlineInfo deadlineBefore = createDeadline(deadlineWindowFrom.minus(1, ChronoUnit.DAYS));
+        ScheduledDeadlineInfo deadlineAfter = createDeadline(deadlineWindowTo.plus(1, ChronoUnit.DAYS));
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Arrays.asList(deadlineBefore, deadlineAfter));
+
+        testSubject.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything());
+    }
+
+    @Test
+    void noDeadlineInTimeframeWithDeadlineInsideWindow() {
+        Instant expiryTime = deadlineWindowFrom.plus(1, ChronoUnit.DAYS);
+        ScheduledDeadlineInfo deadline = createDeadline(expiryTime);
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(deadline));
+
+        assertThrows(AxonAssertionError.class,
+                     () -> testSubject.expectNoScheduledDeadline(deadlineWindowFrom, deadlineWindowTo, deadline.deadlineMessage().getPayload()));
+    }
+
+    @Test
+    void noDeadlineInTimeframeWithDeadlineAtFrom() {
+        ScheduledDeadlineInfo deadline = createDeadline(deadlineWindowFrom);
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(deadline));
+
+        assertThrows(AxonAssertionError.class,
+                     () -> testSubject.expectNoScheduledDeadline(deadlineWindowFrom, deadlineWindowTo, deadline.deadlineMessage().getPayload()));
+    }
+
+    @Test
+    void noDeadlineInTimeframeWithDeadlineAtTo() {
+        ScheduledDeadlineInfo deadline = createDeadline(deadlineWindowTo);
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(deadline));
+
+        assertThrows(AxonAssertionError.class,
+                     () -> testSubject.expectNoScheduledDeadline(deadlineWindowFrom, deadlineWindowTo, deadline.deadlineMessage().getPayload()));
+    }
+
+    @Test
+    void noDeadlineInTimeframeWithDeadlinesOutsideWindow() {
+        ScheduledDeadlineInfo deadlineBefore = createDeadline(deadlineWindowFrom.minus(1, ChronoUnit.DAYS));
+        ScheduledDeadlineInfo deadlineAfter = createDeadline(deadlineWindowTo.plus(1, ChronoUnit.DAYS));
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Arrays.asList(deadlineBefore, deadlineAfter));
+
+        testSubject.expectNoScheduledDeadline(deadlineWindowFrom, deadlineWindowTo, deadlineBefore.deadlineMessage().getPayload());
```

Comment:
Just a thought, but how about we do an `assertDoesNotThrow`?
I think this only makes sense because all the other methods do assert an exception to be thrown.
And, of the nature of the `FixtureExecutionResultImpl`.